src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/Asn1/X501AttributeAsn.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.Asn1;
+using System.Security.Cryptography.X509Certificates;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography.X509Certificates.Asn1
+{
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct X501AttributeAsn
+    {
+        [ObjectIdentifier]
+        internal string AttrId;
+
+        [SetOf]
+        internal AttributeValueAsn[] AttrValues;
+
+        public X501AttributeAsn(X501Attribute attribute, bool copyValue=true)
+        {
+            if (attribute == null)
+            {
+                throw new ArgumentNullException(nameof(attribute));
+            }
+
+            AttrId = attribute.Oid.Value;
+            AttrValues = new[] { new AttributeValueAsn { AttrValue = copyValue ? attribute.RawData.CloneByteArray() : attribute.RawData } };
+        }
+
+        [Choice]
+        internal struct AttributeValueAsn
+        {
+            [AnyValue]
+            internal ReadOnlyMemory<byte>? AttrValue;
+        }
```

Comment:
This feels like quite a cumbersome way to encode it. I need a `SET` with one element that is exactly the value of `RawData` of the `X501Attribute`.